dashboard/modules/reporter/reporter_agent.py

```diff
@@ -650,20 +676,103 @@ def _record_stats(self, stats, cluster_stats):
             raylet_pid = str(raylet_stats["pid"])
             # -- raylet CPU --
             raylet_cpu_usage = float(raylet_stats["cpu_percent"]) * 100
-            raylet_cpu_record = Record(
-                gauge=METRICS_GAUGES["raylet_cpu"],
-                value=raylet_cpu_usage,
-                tags={"ip": ip, "pid": raylet_pid},
+            records_reported.append(
+                Record(
+                    gauge=METRICS_GAUGES["raylet_cpu"],
+                    value=raylet_cpu_usage,
+                    tags={"ip": ip, "pid": raylet_pid},
+                )
             )
 
             # -- raylet mem --
-            raylet_mem_usage = float(raylet_stats["memory_info"].rss) / 1e6
-            raylet_mem_record = Record(
-                gauge=METRICS_GAUGES["raylet_mem"],
-                value=raylet_mem_usage,
-                tags={"ip": ip, "pid": raylet_pid},
+            raylet_mem_full_info = raylet_stats.get("memory_full_info")
+            if raylet_mem_full_info is not None:
+                raylet_uss = float(raylet_mem_full_info.uss) / 1.0e6
+                records_reported.append(
+                    Record(
+                        gauge=METRICS_GAUGES["raylet_mem"],
+                        value=raylet_uss,
+                        tags={"ip": ip, "pid": raylet_pid, "mem_type": "uss"},
+                    )
+                )
+                raylet_rss_other = (
+                    float(raylet_stats["memory_info"].rss) / 1.0e6 - raylet_uss
+                )
+                records_reported.append(
+                    Record(
+                        gauge=METRICS_GAUGES["raylet_mem"],
+                        value=raylet_rss_other,
+                        tags={"ip": ip, "pid": raylet_pid, "mem_type": "rss_other"},
```

Comment:
Can we just report "rss" and then "uss" if it's available? I think this metric will be confusing, and users can also compute it easily from Grafana anyway. I think it's best to report raw metrics and do as little processing here as we can.

Reply:
It is usually desirable to have the metric without any label filter to aggregate to something meanful, e.g. `uss + rss_other = total rss`. Also I think uss is the most interesting metric here. Will update the comment.

Reply:
I think that this is confusing, though. As a user, I need to then understand what `rss_other` means, which is not obvious. On the other hand, everyone should have the same concept of what "rss" means, and users who need to dig deeper can then put in a more complex query.

In general for metrics, I think we always want to report as raw a number as possible to leave it understandable and keep the queries flexible.

Reply:
I think most users don't need to understand `rss_other`, since they can just filter on `mem_type=uss`. I splitted the metric into `workers_mem_rss` and `workers_mem_uss`, where the 2nd metric is only exported when available.

Reply:
Thanks! Can we do the same for raylet: "rss" and optionally "uss" instead of "rss" or ("rss_other" and "uss")?

Reply:
Updated.